ElrondNetwork/elrond-go
# EN-10159: remove old databases at shuffle out

Refactored the deletion of old databases at each epoch start in regards to the node type (validator or observer). It relies on the oldest epoch to keep after iterating over all storage units and wait an epoch for deleting the databases marked for deletion the previous epoch.

Testing scenarios:
- start a testnet with `ValidatorCleanOldEpochsData = true` and `ObserverCleanOldEpochsData = false` inside `config.toml`

Acceptance criteria:
1).
- at any time (even if shuffled out or not), check that old databases of validators are removed
- at any time, check that old databases of observers aren't removed
- at any time, go to one or more validator nodes and set `FullArchive = true` inside `prefs.toml`. old databases should not be removed anymore (this should be noticed in several epochs)

2). 
- set `ValidatorCleanOldEpochsData = false` to a validator and restart it. It should no longer remove old dbs
- set `ObserverCleanOldEpochsData = true` to an observer and restart it. It should start removing old dbs.

3). 
- stake an observer so it becomes validator. Before becoming validator, with the default setting (observers don't clean DB, validators do) databases should not be removed. After the node becomes a validator, check that databases start to be deleted

4). 
- make a validator jailed (easiest way: stop it for 2 epochs when it is a validator). Before being jailed, it was a validator and removed old data. After that, it should not remove anymore
